operations/operation.h

```diff
@@ -40,6 +40,11 @@ class Operation {
     
     // New API for forward Prop
     virtual Tensor<T>* forward() = 0;
+
+    ~Operation() {
+        delete t1;
+        delete t2;
+    }
```

Comment:
What about t3?

Reply:
t3 will be deleted as it's a recursive call. So originally the user will write delete t3;
then the Destructor will go on and call delete t3->backOp.
which will delete t3->backOp->t1 and t3->backOp->t2.

Also, in C++ if you try to delete an already deleted object, it throws an error.